### PR TITLE
*: Update TOC.md and fix some deadlink issues 

### PR DESCRIPTION
**Pull request by yikeke**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -216,219 +229,179 @@
...
+        + [`cluster_load`](/system-tables/system-table-cluster-load.md)
+        + [`cluster_systeminfo`](/system-tables/system-table-cluster-systeminfo.md)
+        + [`cluster_log`](/system-tables/system-table-cluster-log.md)
+        + [`metrics_schema`](/system-tables/system-table-metrics-schema.md)
+        + [`metrics_tables`](/system-tables/system-table-metrics-tables.md)
+        + [`metrics_summary`](/system-tables/system-table-metrics-summary.md)
+        + [`inspection_result`](/system-tables/system-table-inspection-result.md)
+        + [`inspection_summary`](/system-tables/system-table-inspection-summary.md)
+  + UI
+    + TiDB Dashboard
```
Comment (yikeke) [inline TOC.md:344]: Please help update this section after all your dashboard PRs are merged. @TomShawn

Comment (TomShawn) [reply TOC.md:344]: OK

Comment (TomShawn) [reply TOC.md:323]: Updated

File: TOC.md [hunk 2]
```diff
@@ -5,176 +5,191 @@
...
+    + [Hybrid Topology](/hybrid-deployment-topology.md)
+  + Install and Start
+    + Linux OS
+      + [Use TiUP (Recommended)](/production-deployment-using-tiup.md)
+      + [Use TiUP Offline (Recommended)](/production-offline-deployment-using-tiup.md)
+      + [Use TiDB Ansible](/online-deployment-using-ansible.md)
+      + [Use TiDB Ansible Offline](/offline-deployment-using-ansible.md)
+      + From Binary Tarball (Deprecated)
+        + [Production Deployment](/production-deployment-from-binary-tarball.md)
+        + [Testing Deployment](/test-deployment-from-binary-tarball.md)
```
Comment (yikeke) [inline TOC.md:35]:
```suggestion
```

File: TOC.md [hunk 3]
```diff
@@ -260,176 +271,121 @@
...
+      + [Numeric Functions and Operators](/functions-and-operators/numeric-functions-and-operators.md)
+      + [Date and Time Functions](/functions-and-operators/date-and-time-functions.md)
+      + [Bit Functions and Operators](/functions-and-operators/bit-functions-and-operators.md)
+      + [Cast Functions and Operators](/functions-and-operators/cast-functions-and-operators.md)
+      + [Encryption and Compression Functions](/functions-and-operators/encryption-and-compression-functions.md)
+      + [Information Functions](/functions-and-operators/information-functions.md)
+      + [JSON Functions](/functions-and-operators/json-functions.md)
+      + [Aggregate (GROUP BY) Functions](/functions-and-operators/aggregate-group-by-functions.md)
+      + [Window Functions](/functions-and-operators/window-functions.md)
+      + [Miscellaneous Functions]](/functions-and-operators/miscellaneous-functions.md)
```
Comment (lilin90) [inline TOC.md:296]:
```suggestion
      + [Miscellaneous Functions](/functions-and-operators/miscellaneous-functions.md)
```

**Pull request merged**